packages/metro/src/ModuleGraph/worker/collectDependencies.js

```diff
@@ -199,6 +209,16 @@ function collectDependencies<TSplitCondition = void>(
         return;
       }
 
+      if (
```

Comment:
Let's gate this on a new `unstable_allowRequireContext` option that defaults to false.